src/encoder.rs

```diff
@@ -204,17 +204,37 @@ impl<InstructionsType: Instructions> Encoder<InstructionsType> {
             location: _,
         } in &switch.cases
         {
-            // TODO default case is not yet implemented because
-            // the ITE expression is complicated.
-            assert!(literal.is_some());
+            let is_default = literal.is_none();
+
             self.ssa_tracker.set_current_ssa(pre_switch_ssa.clone());
 
             self.encode_block(body);
 
-            let skip_condition = SMTExpression::Not(Box::new(SMTExpression::Eq(
-                Box::new(discriminator[0].name.clone()),
-                self.encode_literal_value(literal.as_ref().unwrap()),
-            )));
+            let skip_condition = if is_default {
+                format!(
+                    "(or {})",
```

Comment:
why or?

Reply:
It's the "skip condition". For every single case it's `(not (eq <discriminator> <case label>))`. So the default is skipped whenever one of the cases is taken. This means `(or (eq <discriminator> <case label>)...)`

Reply:
Ah, should test with a single default case...